.gitignore

```diff
@@ -19,3 +19,6 @@ npm-debug.log*
 
 # Misc
 .DS_Store
+
+# Workspaces
+/.devcontainer/ciphereditor.code-workspace
```

Comment:
I just noticed, you are gitignoring a file that is included in your PR. Which one of the two were you aiming for?

Reply:
It should be like that, but if you want I'll remove that, as the monorepo extension modifies it, but it should be included in the repo, just not tracked for changes

Reply:
As soon as a file gets tracked (which is the case for `ciphereditor.code-workspace` as it is visible here), `.gitignore` does no affect it. So the file should either be included and not ignored or vice-versa. Or am I missing something?

The Visual Studio Code workspace file does change along the way when working with it, right? Maybe it should not be part of version control?

Reply:
Yea that's what I was hoping to do, I recall having done something similar long ago but I think am getting confused. Do you think it might be worth adding a separate template file for tracking, and add a copy command to the dev container postSetup to copy from template to the normal workspace file?

Reply:
I just found that the vscode settings could be moved to `devcontainer.json`. I'd suggest you to do that and remove the workspace file entirely from the project and `.gitignore`.

Reply:
I thought of that but that would create config conflicts when you use the monorepo workspace and not the dev container

Reply:
This way we have a location for project vscode settings and have them active at least in the dev container. Apart from that I would keep it neutral and let people use the editor of their choice.

Reply:
I am still against the use of setting the monorepo settings in the devcontainer config file, but since the workspace file is already in .devcontainer and not .vscode that is fine with me. I'll fix this up real quick and add to the setup script of copying the template workspace file

Reply:
What is wrong with adding the vscode settings in there? Am I missing something?

Reply:
Conventions reasons. As configurations for VSCode pertaining to if the Dev Container is used should be put in the dev container. Settings pertaining to VSCode of VSC Workspaces are used (regardless of usage of Dev Containers) should be in a .code-workspace file. General project settings regardless of the usage of Dev Containers and/or VSC workspace. In this case, since the Monorepo plugin could initiate/generate the workspace configuration, then it's settings should be in the settings.json file in .vscode folder, indeed it was stupid of me originally to embed them into the workspace file, as it was assuming that dev containers were using and the workspace configuration was activated before hand

Reply:
Note that VSC Workspaces and Dev Containers are separate things, In my experience, both in my personal projects and other public projects I have seen VSC Workspace configurations be included alongside Dev Container configurations, as it's easier for a monorepo developer to receive bug reports and PRs that use the same highly reproducible environments.

Reply:
The reason I am restrained here: We're not sponsored by Microsoft and I try to keep the repository as neutral as possible when it comes to the choice of editor people want to use. I'd like to take dev containers to a test run but I don't want to make vscode the 'preferred way' of doing things. I'd rather go for stuff like https://editorconfig.org/, etc. So what I'm really asking is, what is a sensible minimum of vscode specific stuff we can put in to make dev containers work nicely.

Reply:
I mean, I am more than happy to write and maintain other configurations for other editors that I use too if need be, that is the beauty of Open source. It's true that I may have a bias towards this PR, but from my experience, I don't believe that writing extra configurations for a certain editor really is restraining the repo on which editor is best to use, at the end of the day, *anyone* can write and contribute configurations for their editors, this is how the bigger repos work with their community base of contributors.

I took a look at editorconfig, it's a pretty cool idea, but its scope is already handled by ts-standard and the underlying ESLint.